.github/workflows/build.yml

```
@@ -0,0 +1,42 @@
+      - main
+    types:
+      - closed
+
+
+jobs:
+  build-and-deploy:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    environment:
+      name: production
+      url: ${{ steps.deploy.outputs.url }}
```

actionlint (1.7.4)

property "deploy" is not defined in object type {auth: {conclusion: string; outcome: string; outputs: {access_token: string; auth_token: string; credentials_file_path: string; id_token: string; project_id: string}}}

(expression)